src/overlay/test/SurveyMessageLimiterTests.cpp

```diff
@@ -36,12 +36,12 @@ TEST_CASE("messagelimiter", "[overlay][survey][messagelimiter]")
     auto ledgerNum = app->getHerder().trackingConsensusLedgerIndex();
     SurveyRequestMessage firstRequest(v0SecretKey.getPublicKey(),
                                       v1SecretKey.getPublicKey(), ledgerNum,
-                                      temp, SURVEY_TOPOLOGY);
+                                      temp, TIME_SLICED_SURVEY_TOPOLOGY);
```

Comment:
Shouldn't this fail in validation path given that the test doesn't setup the recording phase? Or do we treat the request as valid, but produce an empty response?

Reply:
That test is evaluating `SurveyMessageLimiter`, but [the check for an active survey occurs in `SurveyManager` as a lambda passed in to `SurveyMessageLimiter`](https://github.com/bboston7/stellar-core/blob/66bcfa7cd3a1531e250ed535d30d0336c21f8eb7/src/overlay/SurveyManager.cpp#L318). This helps to keep `SurveyMessageLimiter` small, while `SurveyManager` does the heavier lifting of communicating with other subsystems.

There are tests in `SurveyManagerTests` that check that validation fails for requests outside of the reporting phase.